src/components/RegisterPage.js

```diff
@@ -0,0 +1,190 @@
+                                       'Content-Type': 'application/json',
+                                              },
+                                    body: JSON.stringify({
+                                       "first_name" : this.state.first_name,
+                                       "last_name" : this.state.last_name,
+                                       "user_password" : this.state.confirm_password,
+                                       "mobile_number" : this.state.mobile_number,
+                                       "mobile_verified" : "true",
+                                       "user_role" : "PIGGY_USER",
+                                       "user_email" : this.state.user_email,
+                                       "device_id":"23ADEVIEW"
+                                       })
+                                    })
+        let res = await response.text();
+        if (response.status >= 200 && response.status < 300) {
+           console.log("response from server if --> ", res);
+           this.props.navigation.navigate('Login');
+        } else {
+           console.log("in else block")
+           let error = res;
+           throw error;
+           console.log("error response --> ", error);
+        }
+      } catch(errors) {
+         console.log("inside catch")
+         //let formErrors = JSON.parse(errors);
+         //let errorsArray = [];
+         //for(var key in formErrors){
+           // if(formErrors[key].length > 1) {
+             //  formErrors[key].map(error => errorsArray.push(`${key} ${error}`));
+            //} else {
+              // errorsArray.push(`${key} ${formErrors[key]}`)
+            //}
+         //}
+         console.log("error from server-->> ", errors)
+        this.setState({errors: errors});
+      }
+    }
+
+
+
+    render() {
+      return (
+       <View style = {styles.container}>
+        <View style = {styles.registerform}>
+
+          <TextInput style = {styles.input}
+            placeholder = "enter your first name"
+            returnKeyType = "next"
+            onChangeText = { (text) => this.setState({first_name : text})}
+            onSubmitEditing = {()=> this.secondnameInput.focus()}
+          />
+
+          <TextInput style = {styles.input}
+           placeholder = "enter your second name"
+           returnKeyType = "next"
+           onChangeText = { (text) => this.setState({last_name : text})}
+           onSubmitEditing = {()=> this.phoneInput.focus()}
+            ref = {(input) => this.secondnameInput =input}
+          />
+
+          <TextInput style = {styles.input}
+           placeholder = "enter your phone number"
+           returnKeyType = "next"
+           onChangeText = { (text) => this.setState({mobile_number : text})}
+           onSubmitEditing = {()=> this.emailInput.focus()}
+           ref = {(input) => this.phoneInput =input}
+           />
+
+           <TextInput style = {styles.input}
```

Comment:
Instead of using multiple TextInput components, can create a reusable component which implements TextInput and is initialized with props